src/app/core/internationalization.module.ts

```diff
@@ -2,12 +2,95 @@ import { registerLocaleData } from '@angular/common';
 import { HttpClient } from '@angular/common/http';
 import localeDe from '@angular/common/locales/de';
 import localeFr from '@angular/common/locales/fr';
-import { Inject, LOCALE_ID, NgModule } from '@angular/core';
+import { Inject, Injectable, LOCALE_ID, NgModule } from '@angular/core';
+import { TransferState } from '@angular/platform-browser';
 import { TranslateLoader, TranslateModule, TranslateService } from '@ngx-translate/core';
-import { TranslateHttpLoader } from '@ngx-translate/http-loader';
+import { Observable, OperatorFunction, combineLatest, of } from 'rxjs';
+import { catchError, first, map, switchMap, withLatestFrom } from 'rxjs/operators';
 
-export function translateFactory(http: HttpClient) {
-  return new TranslateHttpLoader(http, 'assets/i18n/', '.json');
+import { SSR_LOCALE, SSR_TRANSLATIONS } from './configurations/state-keys';
+import { StatePropertiesService } from './utils/state-transfer/state-properties.service';
+
+export type Translations = Record<string, string | Record<string, string>>;
+
+export function appendSlash(): OperatorFunction<string, string> {
+  return (source$: Observable<string>) => source$.pipe(map(url => `${url}/`));
+}
+
+export function filterAndTransformKeys(translations: Record<string, string>): Translations {
+  const filtered: Translations = {};
+  for (const key in translations) {
+    if (key.startsWith('pwa-')) {
+      const value = translations[key];
+      try {
+        const parsed = JSON.parse(value);
+        filtered[key.replace('pwa-', '')] = parsed;
```

Comment:
I'd rather use a regex for this replace operation. You never know if that substring is gonna be used at other parts of the key.